src/lib/Components/Artist/Header.tsx

```diff
@@ -46,25 +51,32 @@ class Header extends React.Component<HeaderProps, State> {
     })
   }
 
-  handleFollowChange = () => {
+  @track({ action: "press follow/unfollow button" })
+  handleFollowChange() {
     const newFollowersCount = this.state.following ? this.state.followersCount - 1 : this.state.followersCount + 1
     ARTemporaryAPIModule.setFollowArtistStatus(!this.state.following, this.props.artist._id, (error, following) => {
       if (error) {
         console.error(error)
       } else {
-        Events.postEvent(this, {
-          name: following ? "Follow artist" : "Unfollow artist",
-          artist_id: this.props.artist._id,
-          artist_slug: this.props.artist.id,
-          // TODO At some point, this component might be on other screens.
-          source_screen: "artist page",
-        })
+        this.successfulFollowChange()
       }
       this.setState({ following, followersCount: newFollowersCount })
     })
     this.setState({ following: !this.state.following, followersCount: newFollowersCount })
   }
 
+  // currently you can't get state yet, but leaving this in as desired usage
+  @track((props, state) => ({ name: state.following, artist_id: props.artist._id, artist_slug: props.artist.id }))
```

Comment:
No clue, but no, it does not support state on the methods. At the moment this code doesn't break, but it just omits the field in general. It prints / sends e.g:

`{artist_id: 23940948335, artist_slug: alex-katz}`